neuralmonkey/runners/crf_runner.py

```diff
@@ -0,0 +1,103 @@
+from typing import List, Dict, Set, Optional, Callable
+from typeguard import check_argument_types
+
+from neuralmonkey.logging import log
+from neuralmonkey.model.model_part import ModelPart
+from neuralmonkey.vocabulary import Vocabulary, END_TOKEN_INDEX
+from neuralmonkey.runners.base_runner import (
+    BaseRunner, Executable, FeedDict, ExecutionResult, NextExecute)
+from neuralmonkey.decoders.sequence_labeler import CRFLabeler
+
+# pylint: disable=invalid-name
+Postprocessor = Callable[[List[List[str]]], List[List[str]]]
+# pylint: enable=invalid-name
+
+
+class CRFRunExecutable(Executable):
+
+    def __init__(self,
+                 all_coders: Set[ModelPart],
+                 fetches: FeedDict,
+                 vocabulary: Vocabulary,
+                 postprocess: Optional[Postprocessor]) -> None:
+        self._all_coders = all_coders
+        self._fetches = fetches
+        self._vocabulary = vocabulary
+        self._postprocess = postprocess
+
+        self.result = None  # type: ExecutionResult
+
+    def next_to_execute(self) -> NextExecute:
+        """Get the feedables and tensors to run."""
+        return self._all_coders, self._fetches, None
+
+    def collect_results(self, results: List[Dict]) -> None:
+        if len(results) > 1:
+            raise ValueError("CRFRunner does not support ensembling.")
+
+        loss = results[0].get("loss", 0.)
+        input_mask = results[0]["input_mask"]
+
+        decoded_indices = results[0]["decoded_sequence"]
+
+        assert (input_mask == results[0]["input_mask"]).all()
+
+        # CAUTION! FABULOUS HACK BELIEVE ME
```

Comment:
omg kolik je toho copy-paste? Nedá se z toho vyčlenit dekodovaci funkce, kterou budou sdilet oba runnery?

Reply:
Jo, to by slo. Teoreticky by to mohl byt ciste jen Label Runner s tim, ze ta CRF vrstva by byl jen parametr runneru/dekoderu.